child_compassion/model/child_compassion.py

```diff
@@ -504,13 +547,122 @@ def child_remove_from_typo3(self, cr, uid, ids, context=None):
             with sftp.cd(path):
                 sftp.put(filename)
 
-        self._typo3_scripts_fetch(scripts_url, api_key, "upd_db")
+        return self._typo3_scripts_fetch(scripts_url, api_key, "upd_db")
+
+    def child_add_to_typo3(self, cr, uid, ids, context=None):
+        reload(sys)
+        sys.setdefaultencoding('UTF8')
+
+        # To avoid readding children
+        self.child_remove_from_typo3(cr, uid, ids, context=None)

```

Comment:
This check is useless if we cannot add children that has already the state "On Internet".